app/_hub/kong-inc/exit-transformer/index.md

```diff
@@ -220,24 +360,151 @@ end
+        "kong_message": "No Route matched with those values, arr!"
     }
-    ```
+  ```
+
+### Custom Errors by MIME Type
+
+This example shows a use case where you want custom JSON and HTML responses
+based on an [Accept header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept).
+
+Create a file named `custom-errors-by-mimetype.lua` file with the transformation
+code shown below. See the full list of HTTP response codes [above](#http-msgs).
+Include the status codes you want to customize. Any status code not listed in the
+`custom-errors-by-mimetype.lua` file will use the default
+response `The upstream server responded with <status code>`.
+
+```lua
+local template = require "resty.template"
+local split = require "kong.tools.utils".split
+
+local HTTP_MESSAGES = {
+    s400 = "Bad request",
+    s401 = "Unauthorized",
+    -- ...
+    -- See HTTP Response Status Codes section above for the full list
+    s511 = "Network authentication required",
+    default = "The upstream server responded with %d"
+}
+
+local function get_message(status)
+  return HTTP_MESSAGES["s" .. status] or HTTP_MESSAGES.default.format(status)
+end
+
+local html = template.compile([[
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Some Title</title>
+  </head>
+  <body>
+    <h1>HTTP {{ status }}</h1>
+    <p>{{ error }}</p>
+    <img src="https://thumbs.gfycat.com/RegularJointEwe-size_restricted.gif"/>
+  </body>
+</html>
+]])
+
+-- Customize responses based on content type
+local formats = {
+  ["application/json"] = function(status, message, headers)
+    return status, { status = status, error = message }, headers
+  end,
+  ["text/html"] = function(status, message, headers)
+    return status, html { status = status, error = message }, headers
+  end,
+}
+
+return function(status, body, headers)
+  if status < 400 then
+    return status, body, headers
+  end
+
+  local accept = kong.request.get_header("accept")
+  -- Gets just first accept value. Can be improved to be compliant quality
+  -- etc parser. Look into kong.pdk.response get_response_type
+  if type(accept) == "table" then
+    accept = accept[1]
+  end
+  accept = split(accept, ",")[1]
+
+  if not formats[accept] then
+    return status, body, headers
+  end
+
+  return formats[accept](status, get_message(status), headers)
+end
+```
+
+Configure the `exit-transformer` plugin with `custom-errors-by-mimetype.lua`.
+
+{% navtabs %}
+{% navtab Using cURL %}
+
+```bash
+$ curl -X POST http://<admin-hostname>:8001/services/example.com/plugins \
+ -F "name=exit-transformer"  \
+ -F "config.handle_unknown=true" \
+ -F "config.handle_unexpected=true" \
+ -F "config.functions=@examples/custom-errors-by-mimetype.lua"
+```
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
+
+```bash
+$ http -f :8001/plugins name=exit-transformer \
+  config.handle_unknown=true \
+  config.handle_unexpected=true \
+  config.functions=@examples/custom-errors-by-mimetype.lua
+```
```

Comment:
Comment applies to entire doc: be careful with the indents on code examples. It won't make a difference in the end for whether the command works or not, but for visual consistency+clean copyable code, it's nice to have them follow the same style. As as rule, I've been trying to line things up the way you have for the HTTPie example here: align each line with the start of the command, not the $.

Reply:
> Comment applies to entire doc: be careful with the indents on code examples. It won't make a difference in the end for whether the command works or not, but for visual consistency+clean copyable code, it's nice to have them follow the same style. As as rule, I've been trying to line things up the way you have for the HTTPie example here: align each line with the start of the command, not the $.

@lena-larionova i see what you mean. I just committed reviewer comments in a batch and didn't really notice the difference. I need new glasses.

Reply:
@lena-larionova  I did just notice the discrepancy with the autogenerated examples though. Is there any way we can get those to line up as you suggested for the non-generated examples?

![image](https://user-images.githubusercontent.com/3756245/95138222-5d978300-072f-11eb-9eed-661cf0115996.png)

Reply:
@lena-larionova this PR is ready for another review.